### DISCUSSION TITLE
Correct way to write the path for A1111_HOME variable ?

### DISCUSSION THREAD

**Discussion by BenDes21** [lllyasviel/stable-diffusion-webui-forge#1155] [Q&A] [answered]
Hi there, trying to link my a1111 models to forge, here my webui-user.bat :

```
@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS=

@REM Uncomment following code to reference an existing A1111 checkout.
 set A1111_HOME=C:/Users/Admin/Documents/Stable diffusion/stable-diffusion-webui
 set VENV_DIR=%A1111_HOME%/venv
 set COMMANDLINE_ARGS=%COMMANDLINE_ARGS% ^
 --ckpt-dir %A1111_HOME%/models/Stable-diffusion ^
 --hypernetwork-dir %A1111_HOME%/models/hypernetworks ^
 --embeddings-dir %A1111_HOME%/embeddings ^
 --lora-dir %A1111_HOME%/models/Lora

call webui.bat
```

Issue is that its not load anything, I try to modify the a1111_home with / , // and \ but problem is still here.

Any solution ? ( its maybe a bug but not sure ) 

Thanks

**Comment by anon0730** [comment]
I dunno why it's formatted like that. It didn't work for me on Windows command prompt either. Do this:

```
@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS=

@REM Uncomment following code to reference an existing A1111 checkout.
 set A1111_HOME=C:/Users/Admin/Documents/Stablediffusion/stable-diffusion-webui
 set VENV_DIR=%A1111_HOME%/venv
 set COMMANDLINE_ARGS= --ckpt-dir %A1111_HOME%/models/Stable-diffusion --hypernetwork-dir %A1111_HOME%/models/hypernetworks --embeddings-dir %A1111_HOME%/embeddings --lora-dir %A1111_HOME%/models/Lora

call webui.bat
```

Remove the space from "Stable diffusion" folder name of the original A1111 path to avoid problems (both in the actual folder name and in this ,bat).
Personally I'm not using the `set VENV_DIR=%A1111_HOME%/venv` line either. if you still have any problems try removing it though it will make your Forge folder bigger but still get your models, loras etc. from the A1111 installation.

**Reply by BenDes21** [reply]
> I dunno why it's formatted like that. It didn't work for me on Windows command prompt either. Do this:
> 
> ```
> @echo off
> 
> set PYTHON=
> set GIT=
> set VENV_DIR=
> set COMMANDLINE_ARGS=
> 
> @REM Uncomment following code to reference an existing A1111 checkout.
>  set A1111_HOME=C:/Users/Admin/Documents/Stablediffusion/stable-diffusion-webui
>  set VENV_DIR=%A1111_HOME%/venv
>  set COMMANDLINE_ARGS= --ckpt-dir %A1111_HOME%/models/Stable-diffusion --hypernetwork-dir %A1111_HOME%/models/hypernetworks --embeddings-dir %A1111_HOME%/embeddings --lora-dir %A1111_HOME%/models/Lora
> 
> call webui.bat
> ```
> 
> Remove the space from "Stable diffusion" folder name of the original A1111 path to avoid problems (both in the actual folder name and in this ,bat). Personally I'm not using the `set VENV_DIR=%A1111_HOME%/venv` line either. if you still have any problems try removing it though it will make your Forge folder bigger but still get your models, loras etc. from the A1111 installation.

Thanks a lot for your answer, unfortunately after modify the webui-bat with your correction and rename the stable diffusion folder by Stablediffusion it's still not working :(

<img width="852" alt="ComfyUI_00056_" src="https://github.com/user-attachments/assets/0d3167a8-8110-4d09-a5e0-1afdea5b5ee5">

**Comment by anon0730** [comment]
So you have `set A1111_HOME=C:/Users/Admin/Documents/Stablediffusion/stable-diffusion-webui` in the bat and the A1111 installation is in that exact path? I'm not sure what the problem is if you still can't see your models and loras in Forge.

**Reply by BenDes21** [reply]
> So you have `set A1111_HOME=C:/Users/Admin/Documents/Stablediffusion/stable-diffusion-webui` in the bat and the A1111 installation is in that exact path? I'm not sure what the problem is if you still can't see your models and loras in Forge.

Yes I set everything correctly ( even delete the folder Stablediffusion for shorter the path ) 

<img width="939" alt="Untitled" src="https://github.com/user-attachments/assets/91c63827-4f0c-49e7-9517-6b8f923a0cb9">

**Comment by LeDXIII** [comment]
That's how it works for me:
```

@echo off

set A1111_HOME="D:\SD_New"
set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS=--ckpt-dir %A1111_HOME%/models/Stable-diffusion --hypernetwork-dir %A1111_HOME%/models/hypernetworks --embeddings-dir %A1111_HOME%/embeddings --lora-dir %A1111_HOME%/models/Lora

```
Add double quotation marks around the A1111_HOME value

**Reply by BenDes21** [reply]
> That's how it works for me:
> 
> ```
> 
> @echo off
> 
> set A1111_HOME="D:\SD_New"
> set PYTHON=
> set GIT=
> set VENV_DIR=
> set COMMANDLINE_ARGS=--ckpt-dir %A1111_HOME%/models/Stable-diffusion --hypernetwork-dir %A1111_HOME%/models/hypernetworks --embeddings-dir %A1111_HOME%/embeddings --lora-dir %A1111_HOME%/models/Lora
> ```
> 
> Add double quotation marks around the A1111_HOME value

Yep, did it, my models are still not detected, so strange

<img width="1240" alt="Untitleed" src="https://github.com/user-attachments/assets/e03dafec-9ac4-47b5-9d54-8b271c347ff8">
<img width="1253" alt="Untitled" src="https://github.com/user-attachments/assets/2eee0228-dc47-4a27-9277-a0b1e3cb13c0">

**Reply by LeDXIII** [reply]
> Yep, did it, my models are still not detected, so strange

try replacing / sign in HOME with \ 
and remove space between = and -- here:
`COMMANDLINE_ARGS=--ckpt-dir`

**Reply by BenDes21** [reply]
> > Yep, did it, my models are still not detected, so strange
> 
> try replacing / sign in HOME with \ and remove space between = and -- here: `COMMANDLINE_ARGS=--ckpt-dir`

yep I replaced / by \ and remove the first space and even try with " and without " for a1111_home

I really dont know what's wrong :/

**Reply by LeDXIII** [reply]
> I really dont know what's wrong :/

Maybe the program lacks permissions to the system directory? Try to put some model in a simple path like "D:\model", create a folder there "/models/Stable-diffusion" and change HOME to this value. 

Or just write in --ckpt-dir the path to the standalone model

**Reply by BenDes21** [reply]
> > I really dont know what's wrong :/
> 
> Maybe the program lacks permissions to the system directory? Try to put some model in a simple path like "D:\model", create a folder there "/models/Stable-diffusion" and change HOME to this value.
> 
> Or just write in --ckpt-dir the path to the standalone model

I think the issue is somewhere else, I successfully link the models between ComfyUI and a1111 with the same path

Also, I add git pull into web-user.bat and the log show nothing at the start, Im sure web-user.bat is not even call correctly

**Reply by LeDXIII** [reply]
> Im sure web-user.bat is not even call correctly

This may be an odd solution, but I usually make a copy of the file with the postfix "_1" and put the new file name in "run.bat". 
![image](https://github.com/user-attachments/assets/19dda578-edd4-49c0-943b-8d7659661b8d)
![image](https://github.com/user-attachments/assets/d0f56210-3842-4ddc-b8d5-bf3d3b926b5b)

**Reply by BenDes21** [reply]
> > Im sure web-user.bat is not even call correctly
> 
> This may be an odd solution, but I usually make a copy of the file with the postfix "_1" and put the new file name in "run.bat". ![image](https://private-user-images.githubusercontent.com/177550805/358458411-19dda578-edd4-49c0-943b-8d7659661b8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3NzkxOTgsIm5iZiI6MTcyMzc3ODg5OCwicGF0aCI6Ii8xNzc1NTA4MDUvMzU4NDU4NDExLTE5ZGRhNTc4LWVkZDQtNDljMC05NDNiLThkNzY1OTY2MWI4ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxNlQwMzI4MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMDBlMmRjNTIxNTk2YWFiZDZiNmMwODRlNjA1N2IxMTk5ODEzODY4MTliNTM5YjZlMjk3OWVhMjJiZGM0YTcyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.P9SD8a8DTu54IlggBgT0mr0l0LDcCdnUB5leRb-ySbs) ![image](https://private-user-images.githubusercontent.com/177550805/358458764-d0f56210-3842-4ddc-b8d5-bf3d3b926b5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3NzkxOTgsIm5iZiI6MTcyMzc3ODg5OCwicGF0aCI6Ii8xNzc1NTA4MDUvMzU4NDU4NzY0LWQwZjU2MjEwLTM4NDItNGRkYy1iOGQ1LWJmM2QzYjkyNmI1Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxNlQwMzI4MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ODFmNGFlZWUyMGMzMjk0M2FhZmJhNGI3MzE1NjA0YWVjM2I3YjEzNDA2YTRmNThmNzhhMGVjOTFmNmI2NDQ3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.aNaJWCs8uaBFF0Q17KjRQcBY1quGGTyngNibVFIC__8)

thanks a lot! Gonna try this and then reinstall forge completely, let you know asap

**Reply by BenDes21** [reply, marked as answer]
> > Im sure web-user.bat is not even call correctly
> 
> This may be an odd solution, but I usually make a copy of the file with the postfix "_1" and put the new file name in "run.bat". ![image](https://private-user-images.githubusercontent.com/177550805/358458411-19dda578-edd4-49c0-943b-8d7659661b8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3OTE1NjAsIm5iZiI6MTcyMzc5MTI2MCwicGF0aCI6Ii8xNzc1NTA4MDUvMzU4NDU4NDExLTE5ZGRhNTc4LWVkZDQtNDljMC05NDNiLThkNzY1OTY2MWI4ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxNlQwNjU0MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMjljYzU0NjQ0NzYzMzRjN2I0YzgxYmI0ZmU4OWU4OGM2OTk0MWI2NGE2OGRhNWJkZjdiNDc5NTI2MGY1YWFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.nJVBYvqKFXwnMBL2L3c8IkbKsp79WmBaJlfzLEU5nPo) ![image](https://private-user-images.githubusercontent.com/177550805/358458764-d0f56210-3842-4ddc-b8d5-bf3d3b926b5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3OTE1NjAsIm5iZiI6MTcyMzc5MTI2MCwicGF0aCI6Ii8xNzc1NTA4MDUvMzU4NDU4NzY0LWQwZjU2MjEwLTM4NDItNGRkYy1iOGQ1LWJmM2QzYjkyNmI1Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxNlQwNjU0MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZjM3NTNmMDYyZTllZjQ3MGMyNjU2MDYxODY3N2IxNzM4ODIyMjBmMWMwNGE0ZWUzY2RhMDA1MTdmNzE5OTE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ZV6mUSGKbkIXArJnbGRLgBJLUbdyIB3WtePfqRFjaOw)

alright problem fixed by reinstall completely forge, thanks for your help :)
